### PR TITLE
Add Index.to_frame.

### PR DESCRIPTION
**Pull request by ueshin**

### CODE REVIEW

File: databricks/koalas/tests/test_indexes.py [hunk 1]
```diff
@@ -76,6 +76,9 @@ def test_to_series(self):
         self.assert_eq(kidx.to_series(), pidx.to_series())
         self.assert_eq(kidx.to_series(name='a'), pidx.to_series(name='a'))
 
+        # FIXME: the index values are not addressed the change. (#1190)
+        # self.assert_eq((kidx + 1).to_series(), (pidx + 1).to_series())
```
Comment (ueshin) [inline databricks/koalas/tests/test_indexes.py]: Filed #1190.

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -1312,6 +1383,92 @@ def levshape(self):
...
+
+        sdf = self._internal.sdf.select([scol.alias(name_like_string(idx))
+                                         for scol, idx in zip(self._internal.index_scols, name)]
+                                        + [NATURAL_ORDER_COLUMN_NAME])
+
+        if index:
+            index_map = [(name_like_string(idx), n)
+                         for idx, n in zip(name, self._internal.index_names)]
+        else:
+            index_map = None  # type: ignore
```
Comment (itholic) [inline databricks/koalas/indexes.py]: I'm not pretty sure,

but how about `index_map = [(SPARK_INDEX_NAME_FORMAT(0), None)]`??

then maybe can we avoid `attach_default_index` ?

Comment (ueshin) [reply databricks/koalas/indexes.py]: What's the `SPARK_INDEX_NAME_FORMAT(0)` here then?

Comment (itholic) [reply databricks/koalas/indexes.py]: ah, i was mistaken. sorry for the confusion. 😞

**Pull request merged**